diff --git a/src/Form/AutosaveFormAlterTrait.php b/src/Form/AutosaveFormAlterTrait.php
index dd54e2d..59878a0 100644
--- a/src/Form/AutosaveFormAlterTrait.php
+++ b/src/Form/AutosaveFormAlterTrait.php
@@ -67,7 +67,9 @@ trait AutosaveFormAlterTrait {
       // the message exist and only then offer the options for restore and
       // reject, otherwise will start the auto save process.
       $date = $this->dateFormatter->format($autosave_form_state_timestamp, 'custom', 'M d, Y H:i');
-      $message = $this->t('A version of this page you were editing at @date was saved as a draft. Do you want to resume editing or discard it?', ['@date' => $date]);
+      list($form_id, $entity) = $this->getFormIDandEntity($form_state);
+      $author = $entity->getOwner()->name->value;
+      $message = $this->t('A version of this page you were editing at @date by @author was saved as a draft. Resume or discard it?', ['@date' => $date, '@author' => $author]);
       $form['#attached']['drupalSettings']['autosaveForm']['message'] = (string) $message;
     }
 
