diff --git a/src/Form/EmailTfaVerifyForm.php b/src/Form/EmailTfaVerifyForm.php
index 0c4b599..1f4d1fe 100644
--- a/src/Form/EmailTfaVerifyForm.php
+++ b/src/Form/EmailTfaVerifyForm.php
@@ -67,7 +67,7 @@ class EmailTfaVerifyForm extends FormBase {
   public function buildForm(array $form, FormStateInterface $form_state) {
     $form['email_tfa_verify'] = [
       '#type' => 'textfield',
-      '#title' => $this->t('TFA'),
+      '#title' => $this->t('Enter OTP'),
       '#size' => 60,
       '#required' => TRUE,
     ];
@@ -80,7 +80,7 @@ class EmailTfaVerifyForm extends FormBase {
     $form['interrupt'] = [
       '#type' => 'submit',
       '#name' => 'interrupt',
-      '#value' => $this->t('Interrupt the authentication'),
+      '#value' => $this->t('Cancel'),
       '#submit' => [[$this, 'interruptAuth']],
       '#limit_validation_errors' => [],
       '#button_type' => 'secondary',
