diff --git a/src/Controller/OpignoScormController.php b/src/Controller/OpignoScormController.php
index 363f3ca..dbfc860 100644
--- a/src/Controller/OpignoScormController.php
+++ b/src/Controller/OpignoScormController.php
@@ -21,7 +21,6 @@ class OpignoScormController extends ControllerBase {
     $sco = $scorm_service->scormLoadSco($opigno_scorm_sco);
     // Does the SCO have a launch property ?
     if (!empty($sco->launch)) {
-      $query = [];
 
       // Load the SCO data.
       $scorm = $scorm_service->scormLoadById($sco->scorm_id);
@@ -30,31 +29,12 @@ class OpignoScormController extends ControllerBase {
       if (!empty($sco->attributes['parameters'])) {
         $sco->launch .= $sco->attributes['parameters'];
       }
-      $parts = explode('?', $sco->launch);
-      $launch = array_shift($parts);
-
-      if (!empty($parts)) {
-        // Failsafe - in case a launch URL has 2 or more '?'.
-        $parameters = implode('&', $parts);
-      }
+      $launch = 'scormcontent/index.html';
 
       // Get the SCO location on the filesystem.
       $sco_location = "{$scorm->extracted_dir}/$launch";
       $sco_path = file_create_url($sco_location);
 
-      // Where there any parameters ? If so, prepare them for Drupal.
-      if (!empty($parameters)) {
-        foreach (explode('&', $parameters) as $param) {
-          list($key, $value) = explode('=', $param);
-          $query[$key] = !empty($value) ? $value : '';
-        }
-
-        if ($query) {
-          $query = UrlHelper::buildQuery($query);
-          $sco_path = $sco_path . '?' . $query;
-        }
-      }
-
       return new TrustedRedirectResponse($sco_path);
     }
     else {
