diff --git a/src/Controller/Report/Node.php b/src/Controller/Report/Node.php
index 984bbb5..9ca3089
--- a/src/Controller/Report/Node.php
+++ b/src/Controller/Report/Node.php
@@ -171,7 +171,7 @@ class Node extends ControllerBase {
 
     $page = isset($_GET['page']) ? (int) $_GET['page'] : '';
     $i = 0 + $page * $items_per_page;
-    $timezone =  drupal_get_user_timezone();
+    $timezone =  date_default_timezone_get();
 
     foreach ($results as $data) {
       $user = \Drupal::entityTypeManager()->getStorage('user')->load($data->visitors_uid);
diff --git a/src/Controller/Report/RecentHits.php b/src/Controller/Report/RecentHits.php
index 0d49d28..ecadb67
--- a/src/Controller/Report/RecentHits.php
+++ b/src/Controller/Report/RecentHits.php
@@ -158,7 +158,7 @@ class RecentHits extends ControllerBase {
 
     $page = isset($_GET['page']) ? (int) $_GET['page'] : '';
     $i = 0 + $page * $items_per_page;
-    $timezone =  drupal_get_user_timezone();
+    $timezone =  date_default_timezone_get();
     foreach ($results as $data) {
       $user = \Drupal::entityTypeManager()->getStorage('user')->load($data->visitors_uid);
       $username = array(
diff --git a/src/Form/DateFilter.php b/src/Form/DateFilter.php
index 36c43d2..1418f95
--- a/src/Form/DateFilter.php
+++ b/src/Form/DateFilter.php
@@ -62,7 +62,7 @@ class DateFilter extends FormBase {
 
     $form['visitors_date_filter']['from'] = array(
       '#date_part_order'  => $this->_getOrder(),
-      '#date_timezone'    => drupal_get_user_timezone(),
+      '#date_timezone'    => date_default_timezone_get(),
       '#default_value'    => $from,
       '#element_validate' => array(array($this, 'datelistValidate')),
       '#process'          => array(array($this, 'formProcessDatelist')),
@@ -73,7 +73,7 @@ class DateFilter extends FormBase {
 
     $form['visitors_date_filter']['to'] = array(
       '#date_part_order'  => $this->_getOrder(),
-      '#date_timezone'    => drupal_get_user_timezone(),
+      '#date_timezone'    => date_default_timezone_get(),
       '#default_value'    => $to,
       '#element_validate' => array(array($this, 'datelistValidate')),
       '#process'          => array(array($this, 'formProcessDatelist')),
@@ -263,7 +263,7 @@ class DateFilter extends FormBase {
     $query->addExpression('MIN(visitors_date_time)');
     $min_timestamp = $query->execute()->fetchField();
 
-    $timezone = drupal_get_user_timezone();
+    $timezone = date_default_timezone_get();
 
     return \Drupal::service('date.formatter')->format($min_timestamp, 'custom', 'Y', $timezone);
   }
diff --git a/visitors.info.yml b/visitors.info.yml
index 3bb471f..2ce647b
--- a/visitors.info.yml
+++ b/visitors.info.yml
@@ -4,3 +4,4 @@ package: "Visitors"
 core: "8.x"
 type: module
 configure: visitors.admin
+core_version_requirement: ^8 || ^9
diff --git a/visitors.module b/visitors.module
index bf69436..5b95e01
--- a/visitors.module
+++ b/visitors.module
@@ -67,7 +67,7 @@ function visitors_get_mysql_current_timezone() {
  * @return int
  */
 function visitors_timezone_diff() {
-  $timezone = drupal_get_user_timezone();
+  $timezone = date_default_timezone_get();
   return timezone_offset_get(timezone_open($timezone), date_create());
 }
